src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailedTransportListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\ReceivedStamp;
+use Symfony\Component\Messenger\Stamp\RedeliveryStamp;
+use Symfony\Component\Messenger\Stamp\SentStamp;
+use Symfony\Component\Messenger\Stamp\SentToFailedTransportStamp;
+
+/**
+ * Sends a rejected message to a "failed transport".
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class SendFailedMessageToFailedTransportListener implements EventSubscriberInterface
+{
+    private $messageBus;
+    private $failedSenderAlias;
+    private $logger;
+
+    public function __construct(MessageBusInterface $messageBus, string $failedSenderAlias, LoggerInterface $logger = null)
+    {
+        $this->messageBus = $messageBus;
+        $this->failedSenderAlias = $failedSenderAlias;
+        $this->logger = $logger;
+    }
+
+    public function onMessageFailed(WorkerMessageFailedEvent $event)
+    {
+        if (true === $event->willRetry()) {
+            return;
+        }
+
+        $envelope = $event->getEnvelope();
+
+        // avoid re-sending to the failed sender
+        foreach ($envelope->all(SentStamp::class) as $sentStamp) {
+            /** @var SentStamp $sentStamp */
+            if ($sentStamp->getSenderAlias() === $this->failedSenderAlias) {
+                return;
+            }
+        }
+
+        // remove the received stamp so it's redelivered
+        $throwable = $event->getThrowable();
+        if ($throwable instanceof HandlerFailedException) {
+            $throwable = $throwable->getNestedExceptions()[0];
+        }
+
+        $flattenedException = \class_exists(FlattenException::class) ? FlattenException::createFromThrowable($throwable) : null;
```

Comment:
Using FlattenException was my way of making sure we had an Exception that was serializable. The original exception would be even better (as we already have logic to render this very "pretty" in the console), but I think this is the only safe way.

Reply:
maybe at least pass file+line info when the Debug component is not found?
or maybe remove this and replace by the throwable cast as string?